src/State/README.md

```diff
@@ -0,0 +1,667 @@
+  .execWith(10)
+//=> 400
+
+add(10)
+  .chain(double)
+  .chain(square)
+  .execWith(10)
+//=> 1600
+```
+
+### runWith
+```haskell
+State s a ~> s -> Pair a s
+```
+
+`State` is a lazy datatype that is requires a value for it's state portion to
+be run. A given `State` instance provides a `runWith` method that accepts a
+value to run the instance with. The value must be a member of the type that the
+given `State` instance is fixed to in it's state portion, `s`.
+
+`runWith`, when called with a value, will run the state transition with the given
+value as the initial state and will return the resulting `Pair` with the
+resultant in the `fst` (first) and the state in the `snd` (second).
+
+
+```js
+const State = require('crocks/State')
+const { get, put } = State
+
+const K = require('crocks/combinators/constant')
+
+// swap :: s -> s -> State s s
+const swap = x => old =>
+  put(x)
+    .chain(K(State.of(old)))
+
+//update :: s -> State s s
+const update = x =>
+  get()
+    .chain(swap(x))
+
+update(45)
+  .runWith(100)
+//=> Pair(100, 45)
+```
+
+### evalWith
+```haskell
+State s a ~> s -> a
+```
+
+`State` is a lazy datatype that requires a value for it's state portion to
+be run. A given `State` instance provides an `evalWith` method that accepts a
+value to run the instance with. The value must be a member of the type that the
+given `State` instance is fixed to in it's state portion, `s`.
+
+`evalWith`, when called with a value, will run the state transition with the
+given value as the initial state and will return the resulting resultant and
+will discard the state portion.
+
+```js
+const State = require('crocks/State')
+const { get } = State
+
+const concat = require('crocks/pointfree/concat')
+const flip = require('crocks/combinators/flip')
+const liftA2 = require('crocks/helpers/liftA2')
+const map = require('crocks/pointfree/map')
+const propOr = require('crocks/helpers/propOr')
+
+const name = {
+  first: 'Franklin',
+  last: 'Jennings'
+}
+
+// getLast :: State Object String
+const getFirst =
+  get(propOr('', 'first'))
+
+// getLast :: State Object String
+const getLast =
+  get(propOr('', 'last'))
+
+// inner :: Functor f => f a -> f [ a ]
+const inner =
+  map(Array.of)
+
+// combineNames :: State Object [ String ]
+const combineNames = liftA2(
+  flip(concat),
+  inner(getFirst),
+  inner(getLast)
+)
+
+combineNames
+  .evalWith(name)
+//=> [ 'Franklin', 'Jennings' ]
+```
+
+### execWith
+```haskell
+State s a ~> s -> s
+```
+
+`State` is a lazy datatype that requires a value for it's state portion to
+be run. A given `State` instance provides an `execWith` method that accepts a
+value to run the instance with. The value must be a member of the type that the
+given `State` instance is fixed to in it's state portion, `s`.
+
+`execWith`, when called with a value, will run the state transition with the
+given value as the initial state and will return the resulting state and will
+discard the resultant portion.
```

Comment:
> return the resulting state and will discard the resultant portion.

return the resulting state, discarding the resultant portion.